internal/engine/wazevo/backend/isa/arm64/lower_instr.go

```diff
@@ -425,6 +425,54 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
 		m.lowerVecMisc(vecOpNeg, instr)
 	case ssa.OpcodeVIpopcnt:
 		m.lowerVecMisc(vecOpCnt, instr)
+	case ssa.OpcodeVIshl,
+		ssa.OpcodeVSshr, ssa.OpcodeVUshr:
+		x, y, lane := instr.Arg2WithLane()
+		arr := ssaLaneToArrangement(lane)
+		var modulo byte
+		switch arr {
+		case vecArrangement16B:
+			modulo = 0x7 // Modulo 8.
+		case vecArrangement8H:
+			modulo = 0xf // Modulo 16.
+		case vecArrangement4S:
+			modulo = 0x1f // Modulo 32.
+		case vecArrangement2D:
+			modulo = 0x3f // Modulo 64.
+		default:
+			panic("unsupported lane type " + lane.String())
+		}
+
+		rn := m.getOperand_NR(m.compiler.ValueDefinition(x), extModeNone)
+		rm := m.getOperand_NR(m.compiler.ValueDefinition(y), extModeNone)
+		rd := operandNR(m.compiler.VRegOf(instr.Return()))
+
+		and := m.allocateInstr()
+		and.asALUBitmaskImm(aluOpAnd, rm.nr(), rm.nr(), uint64(modulo), false)
```

Comment:
You should use temporary here as a destination otherwise you have a bug when it's used somewhere (remember my fixe before)